jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpConnection.java

```diff
@@ -627,10 +627,11 @@ public void onFillInterestedFailed(Throwable cause)
     @Override
     public boolean onIdleExpired(TimeoutException timeout)
     {
-        if (_httpChannel.getRequest() == null)
-            return true;
-        ThreadPool.executeImmediately(getExecutor(), _httpChannel.onIdleTimeout(timeout));
-        return false;
+        HttpChannel.IdleTimeoutTask task = _httpChannel.onIdleTimeout(timeout);
```

Comment:
How is the _if request is null, return true_ check done now?

Reply:
The check `getRequest() == null` is kind of equivalent to the previous `isRequestHandled()`, which is now returned in `task.handlingRequest()`.